examples_tests/10.AllocatorTest/main.cpp

```diff
@@ -0,0 +1,438 @@
+};
+
+RandomNumberGenerator rng;
+
+template<typename AlctrType>
+class AllocatorHandler
+{
+	using Traits = core::address_allocator_traits<AlctrType>;
+
+public:
+	void executeAllocatorTest()
+	{
+		const uint32_t testsCnt = rng.getRndAllocCnt();
+		for (size_t i = 0; i < testsCnt; ++i)
+			executeForFrame();
+	}
+
+private:
+	struct AllocationData
+	{
+		uint32_t outAddr = AlctrType::invalid_address;
+		uint32_t size = 0u;
+		uint32_t align = 0u;
+	};
+
+	struct RandParams
+	{
+		uint32_t maxAlign;
+		uint32_t addressSpaceSize;
+		uint32_t alignOffset;
+		uint32_t offset;
+		uint32_t blockSz;
+	};
+
+private:
+	void executeForFrame()
+	{
+		RandParams randAllocParams = getRandParams();
+		void* reservedSpace = nullptr;
+		AlctrType alctr;
+
+		if constexpr (std::is_same<AlctrType, core::LinearAddressAllocator<uint32_t>>::value)
+		{
+			alctr = AlctrType(nullptr, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize);
+		}
+		else
+		{
+			const auto reservedSize = AlctrType::reserved_size(randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+			reservedSpace = _NBL_ALIGNED_MALLOC(reservedSize, _NBL_SIMD_ALIGNMENT);
+			alctr = AlctrType(reservedSpace, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+		}
+
+		// randomly decide how many `multi_allocs` to do
+		const uint32_t multiAllocCnt = rng.getRandomNumber(1u, 500u);
+		for (uint32_t i = 0u; i < multiAllocCnt; i++)
+		{
+			uint32_t addressesToAllcate = allocDataSoA.randomizeAllocData(alctr, randAllocParams);
+			// randomly decide how many allocs in a `multi_alloc` NOTE: must pick number less than `traits::max_multi_alloc`
+
+			Traits::multi_alloc_addr(alctr, addressesToAllcate, allocDataSoA.outAddresses.data(), allocDataSoA.sizes.data(), allocDataSoA.alignments.data());
+
+			// record all successful alloc addresses to the `core::vector`
+			for (uint32_t j = 0u; j < allocDataSoA.size; j++)
+			{
+				if (allocDataSoA.outAddresses[j] != AlctrType::invalid_address)
+					results.push_back({ allocDataSoA.outAddresses[j], allocDataSoA.sizes[j], allocDataSoA.alignments[j] });
+			}
+
+			// run random dealloc function
+			// linear address allocator is always reseted here
+			randFreeAllocatedAddresses(alctr);
+		}
+
+		// randomly choose between reset and freeing all `core::vector` elements
+		if constexpr (!std::is_same<AlctrType, core::LinearAddressAllocator<uint32_t>>::value) //linear address allocator is always reseted in the `randFreeAllocatedAddresses()`
+		{
+			bool reset = static_cast<bool>(rng.getRandomNumber(0u, 1u));
+			if (reset)
+			{
+				alctr.reset();
+				results.clear();
+			}
+			else
+			{
+				// free everything with a series of multi_free
+				while (results.size() != 0u)
+					randFreeAllocatedAddresses(alctr);
+			}
+		}
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/1#discussion_r496264837

> random number of iteration loops (times reset of an allocator is performed)
random number of multi-alloc calls with random number of allocations in the multi-alloc
random number of multi-free calls (not every multi-allocation is paired 1:1 with multi-free)